checker.c

```diff
@@ -1,21 +1,54 @@
 #include <stdio.h>
 #include <assert.h>
+#include<stdbool.h>
+#include "checker.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
+int LowerThresholdChecker(float attribute_value, float lowerthreshold)
+{
+  bool isValid = attribute_value < lowerthreshold;
+  if(isValid)
     return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
+  else return 1;
+}
+
+int UpperThresholdChecker(float attribute_value, float upperthreshold)
+{
+  bool isValid = attribute_value > upperthreshold;
+  if(isValid)
     return 0;
-  }
-  return 1;
+  else return 1;
 }
 
+int batteryIsOk(float temperature, float soc, float chargeRate) {
+  
+  int TemperatureLowerRange,TemperatureHigherRange,TemperatureStatus,SOCLowerRange,SOCHigherRange,SOCStatus,ChargeRateStatus,ValidityofBattery;
+
+  TemperatureLowerRange  = LowerThresholdChecker(temperature,MINTEMP);
```

Comment:
These three lines are very similar for the SoC. Can they be made into a common function?